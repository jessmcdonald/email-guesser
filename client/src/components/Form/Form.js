import React, { Component } from 'react';
import './Form.scss';

class Form extends Component {
  constructor(props){
    super(props);

    this.state = {
      fullNameError : '',
      domainError : '',
      submitError: ''
    };

  }
  handleChange = async (event) => {
    await this.props.onChange(event);
    this.setState({submitError: ''});

    switch(event.target.name) {
      case 'fullName' :
        this.validateFullname();
        break;
      case 'domain' :
        this.validateDomain();
        break;
      default :
        break;
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    if(!this.state.fullNameError && !this.state.domainError) {
      this.props.onSubmit();
    } else {
      this.setState({
        submitError: 'Please check your entries'
      })
    }
  }

  validateFullname = () => {
    let regFullName = /^[a-zA-Z]+ [a-zA-Z]+$/;
    let fullName = this.props.fullName;
    if(!regFullName.test(fullName)) {
      this.setState({
        fullNameError : 'Please enter a full name (first name and last name separated by a space)'
      })
    } else {
      this.setState({
        fullNameError : ''
      })
    }
  }

  validateDomain = () => {
    let domain = this.props.domain;
    if(!domain.includes('.')) {
      this.setState({
        domainError : 'Please enter a valid email domain (e.g. @babbel.com)'
      })
    } else {
      this.setState({
        domainError : ''
      })
    }
  }


  render() {
    return (
      <form className="input-form" onSubmit={this.handleSubmit}>

        <label htmlFor="fname">Full name:</label>
        <input type="text" id="name" name="fullName" placeholder="Jeff Mangum"
          value={this.props.fullName}
          onChange={this.handleChange}>
        </input>
        <div className="fullname-error">
        {this.state.fullNameError ? (
            this.state.fullNameError
          ) : ('')}
        </div>

        <label htmlFor="lname">Domain:</label>
        <input type="text" id="domain" name="domain" placeholder="@babbel.com"
          value={this.props.domain}
          onChange={this.handleChange}>
        </input>
        <div className="domain-error">
        {this.state.domainError ? (
            this.state.domainError
          ) : ('')}
        </div>

        <input type="submit" className="submit-button" value="Submit"></input>

        <div className="submit-error">
        {this.state.submitError ? (
            this.state.submitError
          ) : ('') }
        </div>

      </form>
    );
  }
}

export default Form;