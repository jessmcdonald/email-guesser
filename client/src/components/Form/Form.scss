@import "../../assets/styles/colors.scss";

.input-form {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;

  input, label {
    font-size: 1.4rem;
    padding: 0 0.6rem;
    margin: 0.5rem;
    height: 2rem;
  }

  label {
    width: 10rem;
    background: $pink;
    border-radius: 5px;
    text-align: left;
  }

  input {
    width: 20rem;
    background: none;
    border-bottom: 3px solid $black;
    border-top: none;
    border-left: none;
    border-right: none;
    transition: 0.3s;

    &:hover {
      border-bottom: 3px solid $pink;
    }
  }

  .submit-button {
    background: $lemon;
    color: $black;
    border-radius: 5px;
    border: none;
    height: 3rem;
    margin-top: 2rem;
    font-size: 1.8rem;
    font-weight: 700;
    transition: 0.3s;

    &:hover {
      background: $lilac;
      cursor: pointer;
      border: 3px solid $black;
    }
  }
}