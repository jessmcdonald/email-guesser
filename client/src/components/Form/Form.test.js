import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../Form/Form';

describe("FormComponent", () => {
    it('validates the full name input on change', () => {

    });
    it('validates the full name input on change', () => {

    });
    it('validates the full name input on change', () => {

    });
})