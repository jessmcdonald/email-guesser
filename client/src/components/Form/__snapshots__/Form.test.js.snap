// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<form
  className="input-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="fname"
  >
    Full name:
  </label>
  <input
    id="name"
    name="fullName"
    onChange={[Function]}
    placeholder="Jeff Mangum"
    type="text"
  />
  <div
    className="fullname_error"
  >
    
  </div>
  <label
    htmlFor="lname"
  >
    Domain:
  </label>
  <input
    id="domain"
    name="domain"
    onChange={[Function]}
    placeholder="@babbel.com"
    type="text"
  />
  <div
    className="domain_error"
  >
    
  </div>
  <input
    className="submit-button"
    type="submit"
    value="Submit"
  />
  <div
    className="fullname_error"
  >
    
  </div>
</form>
`;
